### PR TITLE
Military Overhaul: the Tactic Adaptation

### PR DESCRIPTION
**Pull request by zumbak04**
Changelog:
- Adapted vanilla tactics using this pattern to make sure we follow vanilla balance.
  - Casters replace vanilla Pikemen and get modifiers of vanilla Archers.
  - Infantry get modifiers of vanilla Heavy Infantry if there are not vanilla Light Infantry modifiers. If there are, Infantry get average of vanilla Heavy Infantry and vanilla Light Infantry modifiers.
  - Large Infantry get modifiers of vanilla Knights.
  - Aerial Infantry get modifiers of vanilla Knights.
  - Large Cavalry replace vanilla War Elephants.
- Added Horse Archers back.

Secret changelog:
- Created tactic scripted triggers for all unit groups to improve readability.

### CODE REVIEW

File: common/combat_tactics/00_combat_tactics.txt [hunk 1]
```diff
@@ -692,23 +686,25 @@ harras_swarm_tactic = {
...
-				}
-			}
+			light_cavalry_troops_030_tactic_trigger = yes
 		}
 	}
+	
+	# horse_archers
+	horse_archers_offensive = 1.2
+	
+	# light_infantry # get modifiers of archers
```
Comment (DukeJeans) [inline common/combat_tactics/00_combat_tactics.txt]: I'm confus, why not just combine light_infantry and archers into once commented section?

File: common/combat_tactics/00_combat_tactics.txt [hunk 2]
```diff
@@ -717,89 +713,98 @@ harras_swarm_tactic = {
...
 		}
 	}
+	
+	# light_infantry # replaces vanilla pikemen
+	light_infantry_defensive = 2.4
+	archmage_defensive = 2.4
+
+	# heavy_infantry # average of vanilla light_infantry and heavy_infantry
+	heavy_infantry_defensive = 2.1
+	# undead_defensive = 2.1		# No shields
```
Comment (DukeJeans) [inline common/combat_tactics/00_combat_tactics.txt]: I would argue undead are quite capable of using shields, assuming they also wield weapons; they're mindless, but not that mechanically inept. Same with thralls. Swarms is only one that makes sense to me. Mostly depends on how we're classifying Undead and Thralls in terms of actual identity.
image

Comment (zumbak04) [reply common/combat_tactics/00_combat_tactics.txt]: BUT! Swarm got chitin! So... it makes sense to return buff to them all.

File: common/combat_tactics/00_combat_tactics.txt (hunk 1 above)
Comment (zumbak04) [reply common/combat_tactics/00_combat_tactics.txt]: Because there are tactics where archers don't get anything, while light_infantry do.

**Comment by MemeWizard** [comment]
> [trigger.cpp:2192]: Unknown trigger-type: "pikeman" at  file:  line: -1
[trigger.cpp:2192]: Unknown trigger-type: "pikeman" at  file:  line: -1
[trigger.cpp:2192]: Unknown trigger-type: "pikeman" at  file:  line: -1
[trigger.cpp:2192]: Unknown trigger-type: "pikeman" at  file:  line: -1
[trigger.cpp:2192]: Unknown trigger-type: "pikeman" at  file:  line: -1
[trigger.cpp:2192]: Unknown trigger-type: "pikeman" at  file:  line: -1
[trigger.cpp:2192]: Unknown trigger-type: "pikeman" at  file:  line: -1
[trigger.cpp:2192]: Unknown trigger-type: "pikeman" at  file:  line: -1
[trigger.cpp:2192]: Unknown trigger-type: "pikeman" at  file:  line: -1
[trigger.cpp:2192]: Unknown trigger-type: "pikeman" at  file:  line: -1

It's `pikemen`, not `pikeman`

File: common/combat_tactics/00_combat_tactics.txt [hunk 3]
```diff
@@ -1761,36 +1912,46 @@ charging_own_skirmishers_tactic = {
...
+	knights_offensive = 3
+	dragons_offensive = 3
+	
+	# archers
 	archers_defensive = -0.5
-	horse_archers_offensive = -1.2
-	horse_archers_defensive = -0.3
+	# siege_machines_defensive = -0.5	# In back of army
+	archers_offensive = -4.2
+	# siege_machines_offensive = -4.2	# In back of army
```
Comment (MemeWizard) [inline common/combat_tactics/00_combat_tactics.txt]: > In back of army

Just like archers and mages?
Why siege machines don't get offensive debuff? Or why any other melee troops don't get defensive debuff?
You can't shoot things in melee battle direction and hope that your melee troops will remain unharmed.

**Pull request merged**